### PR TITLE
Update Tabs component to support controlled and uncontrolled state

### PR DESCRIPTION
**Pull request by timoheddes**
As outlined in our contributing.md guidelines, our components that require state to function can be either controlled or uncontrolled. This PR updates the Tabs component to support either.

[Asana Link](https://app.asana.com/0/1205288273408141/1205279576909192/f)

### CODE REVIEW

File: packages/libs/react-ui/src/components/Tabs/Tabs.stories.tsx [hunk 1]
```diff
@@ -27,45 +36,67 @@ const meta: Meta<
...
-        <Tabs.Root defaultSelected={defaultSelected}>
-          {items.map((item) => {
+        <Tabs.Root initialTab={initialTab} currentTab={currentTab}>
+          {tabs.map((tab) => {
             return (
-              <Tabs.Tab key={item} value={item}>
-                {item}
+              <Tabs.Tab key={tab} id={tab}>
+                {tab}
               </Tabs.Tab>
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Tabs/Tabs.stories.tsx:92]: This is unrelated to this PR, but do you think we should wrap the tabs in a `Tabs.List`?

**Review by eileenmguo — COMMENTED** [review]
Can you fix dark mode as a part of this PR? 
image

File: packages/libs/react-ui/src/components/Tabs/Tabs.stories.tsx (hunk 1 above)
Comment (timoheddes) [reply packages/libs/react-ui/src/components/Tabs/Tabs.stories.tsx:92]: Yeah I was thinking that. Personally I would opt to refactor the component a bit anyway and rethink its structure a bit. But not as part of this PR.

Comment (eileenmguo) [reply packages/libs/react-ui/src/components/Tabs/Tabs.stories.tsx:92]: Made a ticket to discuss later 👍

**Review by eileenmguo — APPROVED** [review]

**Pull request merged**